pkg/pgxconn/implement.go

```diff
@@ -0,0 +1,102 @@
+package pgxconn
+
+import (
+	"time"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgproto3/v2"
+	"github.com/jackc/pgx/v4"
+)
+
+// rowWithMetric wraps the row returned by QueryRow() for metrics telemetry.
+type rowWithMetric struct {
+	row pgx.Row
+}
+
+func (w rowWithMetric) Scan(dest ...interface{}) error {
+	err := w.row.Scan(dest...)
+	if err != nil && err != pgx.ErrNoRows {
+		errorsTotal.With(promMethodLabel("query_row")).Inc()
+	}
+	return err
+}
+
+// rowsWithDuration wraps the Query() function with duration metric taken to complete the entire execution.
+type rowsWithDuration struct {
+	rows  pgx.Rows
+	start time.Time
+}
+
+func newRowsWithDuration(rows pgx.Rows, start time.Time) rowsWithDuration {
+	return rowsWithDuration{rows, start}
+}
+
+func (r rowsWithDuration) Next() bool {
+	return r.rows.Next()
+}
+
+func (r rowsWithDuration) Scan(dest ...interface{}) error {
+	return r.rows.Scan(dest...)
```

Comment:
Why don't we track these errors? We track them for `rowWithMetric`

Reply:
Because that will cause the errors counter to exceed the total requests counter 😁